starknet_py/net/udc_deployer/deployer.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+
+from typing import Optional, List, Union, cast, NamedTuple
+
+from starkware.starknet.definitions.fields import ContractAddressSalt
+from starkware.starknet.public.abi import get_selector_from_name
+
+from starknet_py.common import int_from_hex
+from starknet_py.constants import DEFAULT_DEPLOYER_ADDRESS
+from starknet_py.net.client_models import Hash, Call
+from starknet_py.net.models import AddressRepresentation, parse_address, compute_address
+from starknet_py.utils.contructor_args_translator import translate_constructor_args
+from starknet_py.utils.crypto.facade import pedersen_hash
+from starknet_py.utils.data_transformer.universal_deployer_serializer import (
+    universal_deployer_serializer,
+    deploy_contract_abi,
+)
+from starknet_py.utils.sync import add_sync_methods
+
+
+ContractDeployment = NamedTuple("ContractDeployment", [("udc", Call), ("address", int)])
+
+
+@add_sync_methods
+class Deployer:
+    """
+    Deployer used to deploy contracts through Universal Deployer Contract (UDC)
+    """
+
+    def __init__(
+        self,
+        *,
+        deployer_address: AddressRepresentation = DEFAULT_DEPLOYER_ADDRESS,
+        account_address: Optional[AddressRepresentation] = None,
+    ):
+        """
+        :param deployer_address: Address of the UDC. Is set to the address of
+            the default UDC (same address on real nets and devnet) by default.
+            Must be set when using custom network other than devnet.
+        :param account_address: Address of an account which will send the transaction.
+            If passed, address of the contract to be deployed will be connected with it.
```

Comment:
Docs should explain how the parameter is used, now it says it's "connected" which makes the explanation a bit cryptic

Reply:
And "Address of an account which will send the transaction" is not completely true, since we can pass a different address and it will still work - right?

Reply:
It will work in a sense that the transaction will pass, but the address returned by the UDC will be different than that of the deployed contract: you'd then have to search for it through a block explorer or something.

Though I agree explanation could be improved a bit.